ios/XMTPModule.swift

```diff
@@ -135,11 +136,21 @@ public class XMTPModule: Module {
 
         // Create a client using its serialized key bundle.
         AsyncFunction("createFromKeyBundle") { (keyBundle: String, environment: String) -> String in
-            let bundle = try PrivateKeyBundle(serializedData: Data(base64Encoded: Data(keyBundle.utf8))!)
-            let options = XMTP.ClientOptions(api: apiEnvironments[environment] ?? apiEnvironments["dev"]!)
-            let client = try await Client.from(bundle: bundle, options: options)
-            self.clients[client.address] = client
-            return client.address
+            do {
+                guard let keyBundleData = Data(base64Encoded: keyBundle),
+                    let bundle = try? PrivateKeyBundle(serializedData: keyBundleData) else {
+                    throw Error.invalidKeyBundle
+                }
+                
+                let options = XMTP.ClientOptions(api: apiEnvironments[environment] ?? apiEnvironments["dev"]!)
+                let client = try await Client.from(bundle: bundle, options: options)
+                self.clients[client.address] = client
+                return client.address
+            } catch {
+                throw error
+                print("ERRO! Failed to create client: \(error)")
+                return ""
```

Comment:
I don't think you'll be able to get to these line of code if your throw on line 150 correct?

Reply:
Indeed. Good catch. Removed and submitted again.

Reply:
I actually think you wanted to keep the throw here instead of the return empty string. It isn't successfully creating a client so it's better to error than to not. The throw fixes the crashing issue by providing information about how to correctly send data to this function. Aka base64 encode it before sending.